packages/core/src/main/mls/MLSService/MLSService.ts

```diff
@@ -59,23 +59,24 @@ export class MLSService {
     return client;
   }
 
-  private async uploadCommitBundle(groupId: Uint8Array, commitBundle: CommitBundle) {
+  private async uploadCommitBundle(groupId: Uint8Array, {commit, welcome}: CommitBundle) {
     const coreCryptoClient = this.getCoreCryptoClient();
 
-    if (commitBundle.welcome) {
-      //@todo: it's temporary - we wait for core-crypto fix to return the actual Uint8Array instead of regular array
-      await this.apiClient.api.conversation.postMlsWelcomeMessage(optionalToUint8Array(commitBundle.welcome));
-    }
-    if (commitBundle.commit) {
+    if (commit) {
       try {
         const messageResponse = await this.apiClient.api.conversation.postMlsMessage(
           //@todo: it's temporary - we wait for core-crypto fix to return the actual Uint8Array instead of regular array
-          optionalToUint8Array(commitBundle.commit),
+          optionalToUint8Array(commit),
         );
         await coreCryptoClient.commitAccepted(groupId);
+        if (welcome) {
```

Comment:
Commit messages need to be sent first only when it's successful can we send the welcome message